packages/core/src/util/cache.js

```diff
@@ -1,16 +1,78 @@
+import { list, tableFromArrays, tableFromIPC, tableToIPC,  uint8, utf8 } from "@uwdata/flechette";
+
 const requestIdle = typeof requestIdleCallback !== 'undefined'
   ? requestIdleCallback
   : setTimeout;
 
+/**
+ * @typedef {import('@uwdata/flechette').Table | Promise<import('@uwdata/flechette').Table>} CacheEntry
+ * @typedef {Object} Cache
```

Comment:
I believe this should be `object`. IIRC `Object` annoyingly has different semantics.